core/src/mindustry/game/Teams.java

```diff
@@ -36,11 +36,19 @@ public Teams(){
 
     @Nullable
     public CoreBuild closestEnemyCore(float x, float y, Team team){
+        CoreBuild closest = null;
+        float closestDst = Float.MAX_VALUE;
+        
         for(Team enemy : team.data().coreEnemies){
-            CoreBuild tile = Geometry.findClosest(x, y, enemy.cores());
-            if(tile != null) return tile;
+            for(CoreBuild core : enemy.cores()){
+                float dst = Mathf.dst2(x, y, core.getX(), core.getY());
+                if(closestDst > dst){
+                    closest = core;
+                    closestDst = dst;
+                }
+            }
```

Comment:
```suggestion
           CoreBuild core = Geometry.findClosest(x, y, enemy.cores());
           float dst = Mathf.dst2(x, y, core.x, core.y);
           if(dst < closestDst){
               closest = core;
               closestDst = dst;
           }
```
Wouldn't this be simpler?

Reply:
perhap (but it will raise NullPointerException since `core` might be null, then you have to add null check which makes it arguably as complicated)

Reply:
~~right, I always forgor the null checks~~